homeassistant/util/unit_conversion.py

```diff
@@ -160,10 +165,17 @@ class TemperatureConverter(BaseUnitConverter):
 
     @classmethod
     def convert(
-        cls, value: float, from_unit: str, to_unit: str, *, interval: bool = False
+        cls,
+        value: float,
+        from_unit: str,
+        to_unit: str,
+        *,
+        bypass_checks: bool = False,
```

Comment:
Why is this necessary? It seems like if it's not supported, how will it ever work?

Reply:
Shouldn't that be just guarded before we call convert function instead of adding a parameter for a single place that uses it ?

Reply:
are we afraid that the checks are going to fail or is it overhead? Because the overhead of 2 in-checks seems minimal.

Reply:
I think it's afraid of overhead, as the call is done inside a loop.
And there are three checks (2 in-checks + 1 instance check to ensure the value is  number)

Reply:
Looking back at the GIT log, this is where the comments were added: https://github.com/home-assistant/core/pull/52386

Reply:
I finally found the relevant note between @thecode and @emontnemery : https://github.com/home-assistant/core/pull/52298/files#r660939298

Reply:
Meh.

Reply:
We can bring back in as a follow-up PR.

Reply:
I'm not very strong leaning either way, it feels a bit pollution of an API for a single use case. Will let others decide. This PR can be merged as it's for a follow up.